src/pages/Home/index.js

```diff
@@ -2,16 +2,134 @@ import Button from 'components/common/Button';
+import { z } from 'zod';
+import Input from 'components/form/Input';
+import ComboBox from 'components/form/ComboBox';
 
 import './styles.css';
+import '../../components/side-bar/style.css';
+import { useEffect, useState } from 'react';
+import { useTopicsMutation } from 'services/topic/topic';
+import SideBar from 'components/side-bar';
+import sideBarIcon from '../../assets/aim.png';
 
 const Home = () => {
   const t = useTranslation();
   const [logout, { isLoading }] = useLogoutMutation();
   const handleLogout = () => logout().then(() => localStorage.removeItem('user'));
+  const [createTarget, { isLoadingCreateTarget, isSuccess, error }] = useCreateTargetMutation();
+  const [getTopics] = useTopicsMutation();
+  const [topics, setTopics] = useState([]);
+  const [latLng, setLatLng] = useState({});
+  const [targets, setTargets] = useState([]);
+  const [getTargets] = useGetTargetMutation();
+
+  const [currentPosition, setCurrentPosition] = useState({
+    ready: false,
+    where: [],
+    error: null,
+  });
+  const geoOptions = {
+    enableHighAccuracy: true,
+    timeout: 2000,
+    maximumAge: 0,
+  };
+  const geolocationEnabled = () => {
+    setCurrentPosition({ ready: false, error: null });
+    return navigator.geolocation.getCurrentPosition(
+      position => {
+        geoSuccess(position);
+      },
+      () => {},
+      geoOptions
+    );
+  };
+
+  const geoSuccess = position => {
+    setCurrentPosition({
+      ready: true,
+      where: [position.coords.latitude, position.coords.longitude],
+    });
+  };
+
+  useEffect(() => {
+    return geolocationEnabled();
+  }, []);
+
+  useEffect(() => {
+    getTopics().then(data => {
+      let topicsPipe = [];
+      data.data.topics.map(topic => {
+        topicsPipe.push(topic.topic);
+      });
+      setTopics(topicsPipe);
+      getAllTargets();
+    });
+  }, []);
+
+  const schema = z.object({
+    radius: z.string().min(1),
+    title: z.string().min(1),
+    topic_id: z.string().min(1),
+  });
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: zodResolver(schema) });
+
+  const onSubmit = data => {
+    createTarget({ ...data, ...latLng })
+      .then(data => {
+        getAllTargets();
+      })
+      .catch(error => {});
+  };
+  const getAllTargets = () => {
+    getTargets().then(data => {
+      const targetList = data.data.targets;
+      for (let i = 0; i < targetList.length; i++) {
+        const newTarget = {
+          lat: targetList[i].target.lat,
+          lng: targetList[i].target.lng,
+        };
+        setTargets(prevState => {
+          return [...prevState, newTarget];
+        });
+      }
+    });
+  };
+  const sendLatLng = dataFromChild => {
+    setLatLng(dataFromChild);
+  };
+
   return (
     <div className="home">
-      <MapView />
+      <MapView currentPosition={currentPosition} sendLatLng={sendLatLng} targets={targets} />
+      <SideBar title={'CREATE TARGET '}>
```

Comment:
It would be a good idea to use translation strings for the titles

Reply:
I've made this changes a few moments ago.